### PR TITLE
feat: [CDS-42950]: Support for Nexus3 Trigger

### PR DESCRIPTION
**Pull request by shivamnegi94**
## Checklist
- [ ] I've documented the changes in the PR description.
- [ ] I've tested this change either in PR or local environment.
- [ ] If hashcheck failed I followed [the checklist](https://harness.atlassian.net/wiki/spaces/DEL/pages/21016838831/PR+Codebasehash+Check+merge+checklist) and updated this PR description with my answers to make sure I'm not introducing any breaking changes.

## Comment Triggers
<details>
  <summary>Build triggers</summary>
  
- Feature build: `trigger feature-build`
- Immutable delegate `trigger publish-delegate`
</details>

<details>
  <summary>PR Check triggers</summary>
  
  You can run multiple PR check triggers by comma separating them in a single comment. e.g. `trigger ti0, ti1`
  
- Compile: `trigger compile`
- CodeFormat: `trigger codeformat`
- MessageMetadata: `trigger messagecheck`
- Recency: `trigger recency`
- BuildNumberMetadata: `trigger buildnum`
- runDockerizationCheck: `trigger dockerizationcheck`
- runAuthorCheck: `trigger authorcheck`
- Checkstyle: `trigger checkstyle`
- PMD: `trigger pmd`
- Feature Name Check: `trigger featurenamecheck`
- TI-bootstrap: `trigger ti0`
- TI-bootstrap1: `trigger ti1`
- TI-bootstrap2: `trigger ti2`
- TI-bootstrap3: `trigger ti3`
- TI-bootstrap4: `trigger ti4`
- FunctionalTest1: `trigger ft1`
- FunctionalTest2: `trigger ft2`
- CodeBaseHash: `trigger codebasehash`
</details>

## PR check failures and solutions
https://harness.atlassian.net/wiki/spaces/BT/pages/21106884744/PR+Checks+-+Failures+and+Solutions

## [Contributor license agreement](https://github.com/harness/harness-core/blob/develop/CONTRIBUTOR_LICENSE_AGREEMENT.md)

- - -
This change is [Reviewable](https://reviewable.io/reviews/harness/harness-core/39261)

### CODE REVIEW

File: 125-cd-nextgen/src/main/java/io/harness/polling/bean/artifact/NexusRegistryArtifactInfo.java [hunk 1]
```diff
@@ -38,13 +48,50 @@ public ArtifactConfig toArtifactConfig() {
       NexusRegistryDockerConfig nexusRegistryDockerConfig =
```
Comment (vtxorxwitty) [inline 125-cd-nextgen/src/main/java/io/harness/polling/bean/artifact/NexusRegistryArtifactInfo.java:48]: Can we take out common setters for all the type here and keep only things which are repository format specific in the if-else.

**Review by vtxorxwitty — COMMENTED** [review]
> Reviewed 1 of 7 files at r1, 9 of 9 files at r2, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/39261)* status: 0 of 1 approvals obtained.

**Review by vtxorxwitty — DISMISSED** [review]
You can ignore the comment if we can't proceed it that way. Approving the PR. 

> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/39261#-:-NFwuomA0g02jRHf9Prm:bitq6tm)* status: :shipit: complete! ✓

File: 125-cd-nextgen/src/main/java/io/harness/polling/bean/artifact/NexusRegistryArtifactInfo.java [hunk 2]
```diff
@@ -38,13 +48,50 @@ public ArtifactConfig toArtifactConfig() {
...
           .build();
+    } else if (repositoryFormat.equalsIgnoreCase("maven")) {
+      NexusRegistryMavenConfig nexusRegistryMavenConfig =
+          NexusRegistryMavenConfig.builder()
+              .artifactId(ParameterField.<String>builder().value(artifactId).build())
+              .groupId(ParameterField.<String>builder().value(groupId).build())
+              .classifier(ParameterField.<String>builder().value(classifier).build())
+              .extension(ParameterField.<String>builder().value(extension).build())
+              .build();
+      return NexusRegistryArtifactConfig.builder()
```
Comment (puthrayaharness) [inline 125-cd-nextgen/src/main/java/io/harness/polling/bean/artifact/NexusRegistryArtifactInfo.java:68]: We can refactor this code. NexusRegistryArtifactConfig can move down.
Have a top-level variable for spec. Just initialize the spec based on repo format

**Review by puthrayaharness — CHANGES_REQUESTED** [review]

**Review by puthrayaharness — APPROVED** [review]

**Comment by shivamnegi94** [comment]
trigger compile

**Review by vaibhav-si — APPROVED** [review]
> Reviewed 3 of 9 files at r2, 41 of 41 files at r3, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/39261)* status: :shipit: complete! ✓

**Review by abhinavmittal-harness — APPROVED** [review]
> Reviewed 3 of 9 files at r2, 41 of 41 files at r3, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/39261)* status: :shipit: complete! ✓

**Review by deepak-harness — APPROVED** [review]
> Reviewed 3 of 9 files at r2, 41 of 41 files at r3, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/39261)* status: :shipit: complete! ✓

**Pull request merged**